### ISSUE TITLE
Add option to enforce TLSV1.2

### ISSUE THREAD

**Issue by The-Nutty** [haraka/Haraka#2708]
### Expected behavior
There should be a way to enforce using TLSV1.2.
### Observed behavior
Currently for some reason haraka is currently negitiating with clients to use. 
The following is from a connection.tls object in the data_post hock.
```
    "cipher": {
        "name": "ECDHE-RSA-AES128-GCM-SHA256",
        "version": "TLSv1/SSLv3"
    },
```

Further more i am seeing the following header in emails
```
Received: from sending-server (mail.sending-server.domain [1.1.1.1])

	by my-server.domain (Haraka/2.8.24) with ESMTPS id 07416364-7082-4B6B-A773-581B2714D96B.1

	envelope-from <me@gmail.com>

	(version=TLSv1/SSLv3 cipher=ECDHE-RSA-AES128-GCM-SHA256 verify=FAIL);
```
### Steps to reproduce
Use haraka with the tls plugin
```
key=key.pem
cert=cert.pem
```
I have also tried adding the following to no luck 
```
secureProtocol=TLSv1_2_method
ciphers=ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA256:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!SRP:!CAMELLIA
```

**Comment by msimerson** [comment]
Fixed in #2648 and released in Haraka 2.8.25.

**Comment by danieltigse** [comment]
quick question. So If I use `secureProtocol=TLSv1_2_method` am I enforcing the connection to use TLS V1.2?
